packages/util/src/url.ts

```diff
@@ -1,27 +1,46 @@
-export function join(...segments: string[]) {
-    const parts: string[] = segments.reduce((parts, segment) => {
-        // Remove leading slashes from non-first part.
-        if (parts.length > 0) {
-            segment = segment.replace(/^\//, "");
-        }
-        // Remove trailing slashes.
-        segment = segment.replace(/\/$/, "");
-        return [...parts, ...segment.split("/")];
-    }, []);
-    const resultParts = [];
-    for (const part of parts) {
-        if (part === ".") {
-            continue;
-        }
-        if (part === "..") {
-            resultParts.pop();
-            continue;
+type posixSeparator = "/";
+type win32Separator = "\\";
+
+function url(separator: posixSeparator | win32Separator) {
+
+    function join(...segments: string[]) {
+        const parts: string[] = segments.reduce((parts, segment) => {
+            // Remove leading slashes from non-first part.
+            if (parts.length > 0) {
+                segment = segment.replace(separator === "\\" ? /^\\/ : /^\//, "");
+            }
+            // Remove trailing slashes.
+            segment = segment.replace(separator === "\\" ? /\\$/ : /\/$/, "");
+            return [...parts, ...segment.split(separator)];
+        }, []);
+
+        const resultParts = [];
+        for (const part of parts) {
+            if (part === ".") {
+                continue;
+            }
+            if (part === "..") {
+                resultParts.pop();
+                continue;
+            }
+            resultParts.push(part);
         }
-        resultParts.push(part);
+        return resultParts.join(separator);
+    }
+
+    function dirname(path: string) {
```

Comment:
Reminder to check the naming convention & functionality of this vs Node

Reply:
I added some tests for this and indeed there are some discrepancies...  

I think the solution is to take https://github.com/browserify/path-browserify and update it with the latest LTS NodeJS version?